Dev/VideoChatClient/Input/ImageInput.cpp

```diff
@@ -0,0 +1,27 @@
+#include "ImageInput.hpp"
+
+#include "VideoChatClientException/VideoChatClientException.hpp"
+
+ImageInput::ImageInput() : m_capture()
+{
+	m_capture.open(cv::CAP_ANY);
+	if (!m_capture.isOpened())
+	{
+		throw VideoChatClientException(VideoChatClientStatus::IMAGE_INPUT_OPEN_CAPTURE_FAILED, 0); //TODO
+	}
+	m_capture.set(cv::CAP_PROP_FRAME_WIDTH, IMAGE_WIDTH);
+	m_capture.set(cv::CAP_PROP_FRAME_HEIGHT, IMAGE_HEIGHT);
+}
+
+void ImageInput::take_input()
+{
+	cv::Mat frame{};
+
+	m_capture.read(frame);
+
+	if (frame.empty())
+	{
+		throw VideoChatClientException(VideoChatClientStatus::IMAGE_INPUT_READ_FRAME_FAILED, 0); //TODO
+	}
+	cv::imencode(".jpg", frame, m_data, std::vector<int32_t>());
```

Comment:
I don't think that passing "std::vector<int32_t>()" is what you should pass.
it should be used for extra flags like the quality of the image and you are passing nothing...
read the docs of it👀

Reply:
Actually, that's how you're supposed to do it. Just saw that it also has a default value
![image](https://user-images.githubusercontent.com/46401832/180446660-1de34a38-266b-4c98-8dc2-328d89ce97e8.png)